common/BitUtils.h

```diff
@@ -84,6 +103,32 @@ namespace Common
 		// Perform our count leading zero.
 		return std::countl_zero(static_cast<u32>(n));
 	}
+
+	__fi static s32 clz(s32 x)
+	{
+		x |= x >> 1;
+		x |= x >> 2;
+		x |= x >> 4;
+		x |= x >> 8;
+		x |= x >> 16;
+
+		return debruijn32[(u32)x * 0x8c0b2891u >> 26];
+	}
+
+	__fi static s32 BitScanReverse8(s32 b)
+	{
+		return msb[b];
+	}
+
+	__fi static s32 GetMostSignificantBitPosition(u32 value)
+	{
+		for (s32 i = 31; i >= 0; i--)
+		{
+			if (((value >> i) & 1) != 0)
+				return i;
+		}
+		return -1;
```

Comment:
```suggestion
#ifdef _MSC_VER
		unsigned long bit;
		return _BitScanReverse(&bit, value) ? bit : -1;
#else
		return value ? __builtin_clz(value) ^ 31 : -1;
#endif
```
And you can drop the rest of the changes and make BitScanReverse8 and clz just call this (or pick one stop using the rest).
As a side note, not handling the value == 0 case will be a bit cheaper, and in my experience in pretty much all cases, the caller has already checked and is only calling this if they know `value` is nonzero, so the separate check here is just wasted cycles.

Reply:
Ok, I will do that when possible and push, I admit I lacked experience with the BitScan part.